src/wallet/spend.cpp

```diff
@@ -1134,20 +1143,34 @@ bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet,
     }
     wallet.chain().findCoins(coins);
 
+    // Check if any input was already spent by another tx in the mempool
+    bool is_replacement = false;
     for (const CTxIn& txin : tx.vin) {
         const auto& outPoint = txin.prevout;
         if (wallet.IsMine(outPoint)) {
             // The input was found in the wallet, so select as internal
             coinControl.Select(outPoint);
+            // Check if was spent
+            bool in_mempool = false;
+            if (wallet.IsSpent(outPoint, &in_mempool)) is_replacement = in_mempool;
         } else if (coins[outPoint].out.IsNull()) {
             error = _("Unable to find UTXO for external input");
             return false;
         } else {
             // The input was not in the wallet, but is in the UTXO set, so select as external
             coinControl.SelectExternal(outPoint, coins[outPoint].out);
+            // Check if was spent in any mempool tx
+            if (!is_replacement) is_replacement = wallet.chain().isSpentInMempool(outPoint);
         }
     }
 
+    // If this tx is a replacement, disallow unconfirmed UTXO selection by rising the
+    // coin control min depth param. Ensuring that we don't add any new unconfirmed UTXOs
+    // to replacement transactions (satisfying BIP125 rule 2).
```

Comment:
please do not add new mentions of BIP125 that aren't signaling-related, see https://github.com/bitcoin/bitcoin/pull/25775